### PR TITLE
Remove Oraclize

### PR DESCRIPTION
**Pull request by riccardopersiani**
Removes Oraclize in favour of CryptoCompare.

### CODE REVIEW

**Review by nostdm — APPROVED** [review]

File: contracts/externals/ECDSA.sol [hunk 1]
```diff
@@ -67,6 +67,37 @@ library ECDSA {
...
+            revert("ECDSA: invalid signature 'v' value");
+        }
+
+        // If the signature is valid (and not malleable), return the signer address
+        address signer = ecrecover(hash, v, r, s);
+        require(signer != address(0), "ECDSA: invalid signature");
+
+        return signer;
+    }
+
```
Comment (mischat) [inline contracts/externals/ECDSA.sol:100]: hey, I know that I am a bit late to the game here, but i will review anyways....

Where did this function come from ?

File: contracts/oracle.sol [hunk 2]
```diff
@@ -21,35 +21,27 @@
...
 import "./externals/strings.sol";
 import "./internals/controllable.sol";
 import "./internals/date.sol";
 import "./internals/ensResolvable.sol";
 import "./internals/parseIntScientific.sol";
 import "./internals/tokenWhitelistable.sol";
-import "./internals/transferrable.sol";
 
 /// @title Oracle provides asset exchange rates and conversion functionality.
-contract Oracle is ENSResolvable, usingProvable, Transferrable, Base64, Date, Controllable, ParseIntScientific, TokenWhitelistable {
```
Comment (mischat) [inline contracts/oracle.sol:35]: Why did we chose to ditch the "transferrable" and ultimately ditch the Claim method ?

File: contracts/oracle.sol [hunk 3]
```diff
@@ -85,64 +74,26 @@ contract Oracle is ENSResolvable, usingProvable, Transferrable, Base64, Date, Co
...
-
-    /// @notice Withdraw tokens from the smart contract to the specified account.
-    function claim(
-        address payable _to,
-        address _asset,
-        uint256 _amount
-    ) external onlyAdmin {
-        _safeTransfer(_to, _asset, _amount);
-        emit Claimed(_to, _asset, _amount);
-    }
```
Comment (mischat) [inline contracts/oracle.sol:128]: as above, why ditch the claim method ?

File: contracts/oracle.sol [hunk 4]
```diff
@@ -152,25 +103,23 @@ contract Oracle is ENSResolvable, usingProvable, Transferrable, Base64, Date, Co
...
 
-            _updateTokenRate(token, parsedRate, parsedLastUpdate);
+            _updateTokenRate(_token, parsedRate, parsedLastUpdate);
         }
     }
 
-    /// @notice Extracts JSON rate value from the response object.
+    /// @dev Extracts JSON rate value from the response object.
     /// @param _json body of the JSON response from the CryptoCompare API.
-    function _parseRate(string memory _json) internal pure returns (string memory) {
```
Comment (mischat) [inline contracts/oracle.sol:167]: I thought we name internal / private functions with an underscore, why did we change the method name ?

File: contracts/oracle.sol [hunk 5]
```diff
@@ -317,24 +218,23 @@ contract Oracle is ENSResolvable, usingProvable, Transferrable, Base64, Date, Co
...
+    ) private pure returns (bool) {
         address signer;
-        bool signatureOK;
-
         // Checks if the signature is valid by hashing the headers
-        (signatureOK, signer) = ecrecovery(sha256(_headers), _signature);
-        return signatureOK && signer == address(uint160(uint256(keccak256(_publicKey))));
+        bytes32 dataHash = sha256(_headers);
+        signer = dataHash.recoverNonMalleable(_signature);
+        return signer == address(uint256(keccak256(_publicKey)));
```
Comment (mischat) [inline contracts/oracle.sol:234]: this is a lot neater ...

File: pkg/bindings/mocks/isValidSignatureExporter.go [hunk 6]
```diff
@@ -31,7 +31,7 @@ var (
...
...03357600080fd5b5051600080546001600160a01b039092166001600160a01b031990921691909117905561021f806100656000396000f3fe608060405234801561001057600080fd5b506004361061002b5760003560e01c806320c13b0b14610030575b600080fd5b6100f26004803603604081101561004657600080fd5b81019060208101813564010000000081111561006157600080fd5b82018360208201111561007357600080fd5b8035906020019184600183028401116401000000008311171561009557600080fd5b9193909290916020810190356401000000008111156100b357600080fd5b8201836020820111156100c557600080fd5b803590602001918460018302840111640100000000831117156100e757600080fd5b50909250905061010f565b604080516001600160e01b03199092168252519081900360200190f35b60008054604080516320c13b0b60e01b815260048101918252604481018790526001600160a01b03909216916320c13b0b9188918891889188919081906024810190606401878780828437600083820152601f01601f191690910184810383528581526020019050858580828437600081840152601f19601f820116905080830192505050965050505050505060206040518083038186803b1580156101b457600080fd5b505afa1580156101c8573d6000803e3d6000fd5b505050506040513d60208110156101de57600080fd5b50519594505050505056fea26469706673582212204f9b44c63ccb4604230e8685c42eca94a9b4cb70cd336ee439f1669c9350e3d464736f6c634300060c0033"
```
Comment (mischat) [inline pkg/bindings/mocks/isValidSignatureExporter.go:34]: what changed here, i can't tell ?

File: test/oracle/oracle_suite_test.go [hunk 7]
```diff
@@ -95,7 +78,7 @@ var _ = AfterEach(func() {
 })
 
 var _ = AfterSuite(func() {
-	TestRig.ExpectMinimumCoverage("oracle.sol", 100.00)
+	TestRig.ExpectMinimumCoverage("oracle.sol", 98.00)
```
Comment (mischat) [inline test/oracle/oracle_suite_test.go:81]: :) What happened here, are the last 2% not testable ?

**Review by mischat — COMMENTED** [review]
I thought I would leave some comments here :)

**Pull request merged**